builder/dockerfile/evaluator.go

```diff
@@ -56,10 +58,32 @@ var allowWordExpansion = map[string]bool{
 	command.Expose: true,
 }
 
-var evaluateTable map[string]func(*Builder, []string, map[string]bool, string) error
+type dispatchRequest struct {
+	builder    *Builder // TODO: replace this with a smaller interface
```

Comment:
comment: Rather than smaller interface, I think this should become a typed/normalized definition struct of the whole build. Then we can run preprocessing steps on it, for example, remove build stages that are not needed and execute the rest.